pkg/iosrc/uri.go

```diff
@@ -0,0 +1,87 @@
+package iosrc
+
+import (
+	"errors"
+	"net/url"
+)
+
+type URI struct {
+	url.URL
+}
+
+const (
+	stdin  = "stdio:///stdin"
+	stdout = "stdio:///stdout"
+	stderr = "stdio:///stderr"
+)
+
+// ParseURI parses the path using `url.Parse`. If the provided uri does not
+// contain a scheme, the scheme will set to file. Relative paths will be
+// treated as files and resolved as absolute paths using filepath.Abs.
+// path is an empty, Scheme is set to file.
+func ParseURI(path string) (URI, error) {
```

Comment:
Maybe return a *URI? (url.Parse returns a *url.URL.)

Reply:
Not a fan. I want this to be immutable like time.Time.

Reply:
Ok uri would have to be hidden in the struct for it to be totally immutable but I like the fact that I can pass the default return value here into a function and know it will not me mutated.